spec/models/vm_or_template_spec.rb

```diff
@@ -471,4 +471,34 @@
       expect(VmOrTemplate.batch_operation_supported?(:migrate, [vm1.id, vm2.id])).to eq(true)
     end
   end
+
+  context ".set_tenant_from_group" do
```

Comment:
Maybe add before validation or something better than I can think of to the description here?  Just glancing at the tests, none of them call this method directly so you'd have to look at the code to realize that just creating tenant will call this method through validations.